### PR TITLE
sgp replace constraint substitutions with past subs

### LINKED ISSUE #1207: linked substitutions replace their function values in .substitutions after a solve
convexengineering/ondemand@77bc3b2

```python
In [1]: from stol import Mission

In [2]: M = Mission(sp=True)

In [3]: M.cost = M["S_{runway}"]*M[M.aircraft.topvar("W")]/M["R"]  # substitutions here include lambda fns

In [4]: M.localsolve("mosek")  # substitutoins after this fails do not
```

### PR DESCRIPTION
**Pull request by mjburton11**
Fixes #1207

### CODE REVIEW

**Comment by bqpd** [comment]
add to whitelist pls

**Comment by mjburton11** [comment]
test models please

**Comment by bqpd** [comment]
test this please

File: gpkit/tests/t_sub.py [hunk 1]
```diff
@@ -382,6 +382,19 @@ def test_vector_sub(self):
...
+        y = Variable("y", lambda c: 2*c[x])
+        z = Variable("z")
+        w = Variable("w")
+
+        with SignomialsEnabled():
+            cnstr = [z + w >= y*x, w <= y]
+
+        m = Model(z, cnstr)
+        m.localsolve("mosek")
+        assert hasattr(m.substitutions["y"], "__call__")
```
Comment (bqpd) [inline gpkit/tests/t_sub.py]: don't use assert! use self.assertTrue

Comment (mjburton11) [reply gpkit/tests/t_sub.py]: done

**Comment by mjburton11** [comment]
test this please

**Comment by bqpd** [comment]
test models please

**Comment by bqpd** [comment]
test this please

**Comment by bqpd** [comment]
@galbramc oh wait now they are. weird.

**Pull request merged**